### PR TITLE
feat(Geo): add updateLocation and deleteLocationHistory api

### PR DESCRIPTION
**Pull request by phantumcode**
*Issue #, if available:*

*Description of changes:*
* add updateLocation and updateLocationHistory api to Geo category

*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Geo/AWSLocationGeoPluginTests/AWSLocationGeoPluginClientBehaviorTests.swift [hunk 1]
```diff
@@ -241,4 +241,91 @@ class AWSLocationGeoPluginClientBehaviorTests: AWSLocationGeoPluginTestBase {
...
+            let device = Geo.Device.unchecked(id: "123-456-789")
+            let location = Geo.Location(latitude: 44.4, longitude: 45.5)
+            try await geoPlugin.updateLocation(location, for: device, with: Geo.UpdateLocationOptions())
+            XCTFail("This call returned success when a failure was expected.")
+        } catch {
+            guard let geoError = error as? Geo.Error else {
+                XCTFail("Error thrown should be Geo.Error")
+                return
+            }
+            XCTAssertEqual(geoError.errorDescription, "No tracker available.")
```
Comment (thisisabhash) [inline AmplifyPlugins/Geo/AWSLocationGeoPluginTests/AWSLocationGeoPluginClientBehaviorTests.swift:285]: suggestion: `XCTAssertEqual(geoError.errorDescription, GeoPluginErrorConstants.missingTracker.errorDescription)`

File: AmplifyPlugins/Geo/AWSLocationGeoPluginTests/AWSLocationGeoPluginClientBehaviorTests.swift [hunk 2]
```diff
@@ -241,4 +241,91 @@ class AWSLocationGeoPluginClientBehaviorTests: AWSLocationGeoPluginTestBase {
...
+        do {
+            let device = Geo.Device.unchecked(id: "123-456-789")
+            try await geoPlugin.deleteLocationHistory(for: device, with: Geo.DeleteLocationOptions())
+            XCTFail("This call returned success when a failure was expected.")
+        } catch {
+            guard let geoError = error as? Geo.Error else {
+                XCTFail("Error thrown should be Geo.Error")
+                return
+            }
+            XCTAssertEqual(geoError.errorDescription, "No tracker available.")
```
Comment (thisisabhash) [inline AmplifyPlugins/Geo/AWSLocationGeoPluginTests/AWSLocationGeoPluginClientBehaviorTests.swift:328]: suggestion: `XCTAssertEqual(geoError.errorDescription, GeoPluginErrorConstants.missingTracker.errorDescription)`

File: AmplifyPlugins/Geo/AWSLocationGeoPlugin/Configuration/AWSLocationGeoPluginConfiguration.swift [hunk 3]
```diff
@@ -53,24 +54,34 @@ public struct AWSLocationGeoPluginConfiguration {
                 throw GeoPluginConfigError.searchDefaultNotFound(indexName: defaultSearchIndex)
             }
         }
+        
+        let trackerConfigJSON = configObject[Section.tracker.key]
```
Comment (thisisabhash) [inline AmplifyPlugins/Geo/AWSLocationGeoPlugin/Configuration/AWSLocationGeoPluginConfiguration.swift:58]:
```
"trackers": { 
       "items": ["tracker823xyz", "tracker725cba"],
       "default": "tracker823xyz"
} 
```
i saw this in the approved CLI contract. do we need to store default and other trackers separately?

File: AmplifyPlugins/Geo/AWSLocationGeoPlugin/Configuration/AWSLocationGeoPluginConfiguration+Keys.swift [hunk 4]
```diff
@@ -23,6 +23,7 @@ extension AWSLocationGeoPluginConfiguration {
         case plugin
         case maps
         case searchIndices
+        case tracker
```
Comment (thisisabhash) [inline AmplifyPlugins/Geo/AWSLocationGeoPlugin/Configuration/AWSLocationGeoPluginConfiguration+Keys.swift:26]:
```
"trackers": { 
          "items": ["tracker823xyz", "tracker725cba"],
           "default": "tracker823xyz"
 } 
```
should this key be `trackers` ?

File: AmplifyPlugins/Geo/AWSLocationGeoPlugin/Dependency/AWSLocationAdapter.swift [hunk 5]
```diff
@@ -40,4 +40,12 @@ class AWSLocationAdapter: AWSLocationBehavior {
     func searchPlaceIndex(forPosition: SearchPlaceIndexForPositionInput) async throws -> SearchPlaceIndexForPositionOutputResponse {
         return try await location.searchPlaceIndexForPosition(input: forPosition)
     }
+    
+    public func updateLocation(forUpdateDevicePosition: BatchUpdateDevicePositionInput) async throws -> BatchUpdateDevicePositionOutputResponse {
```
Comment (thisisabhash) [inline AmplifyPlugins/Geo/AWSLocationGeoPlugin/Dependency/AWSLocationAdapter.swift:44]: should this be `public` function?

File: Amplify/Categories/Geo/DeviceTracking/Geo+DeleteLocationOptions.swift [hunk 6]
```diff
@@ -0,0 +1,18 @@
+//
+//  File.swift
+//  
+//
+//  Created by Pham, Tuan on 9/26/22.
```
[the remark this answers is not in the document]
Comment (phantumcode) [reply Amplify/Categories/Geo/DeviceTracking/Geo+DeleteLocationOptions.swift]: Header updated.

File: AmplifyPlugins/Geo/AWSLocationGeoPluginTests/AWSLocationGeoPluginClientBehaviorTests.swift (hunk 1 above)
Comment (phantumcode) [reply AmplifyPlugins/Geo/AWSLocationGeoPluginTests/AWSLocationGeoPluginClientBehaviorTests.swift]: Updated per suggestion.

File: AmplifyPlugins/Geo/AWSLocationGeoPluginTests/AWSLocationGeoPluginClientBehaviorTests.swift (hunk 2 above)
Comment (phantumcode) [reply AmplifyPlugins/Geo/AWSLocationGeoPluginTests/AWSLocationGeoPluginClientBehaviorTests.swift]: Updated per suggestion

File: AmplifyPlugins/Geo/AWSLocationGeoPlugin/Configuration/AWSLocationGeoPluginConfiguration.swift (hunk 3 above)
Comment (phantumcode) [reply AmplifyPlugins/Geo/AWSLocationGeoPlugin/Configuration/AWSLocationGeoPluginConfiguration.swift]: Good point, I've updated the Geo configuration to match the cli contract.

File: AmplifyPlugins/Geo/AWSLocationGeoPlugin/Dependency/AWSLocationAdapter.swift (hunk 5 above)
Comment (phantumcode) [reply AmplifyPlugins/Geo/AWSLocationGeoPlugin/Dependency/AWSLocationAdapter.swift]: Fixed

File: AmplifyPlugins/Geo/AWSLocationGeoPlugin/Configuration/GeoPluginConfigError.swift [hunk 7]
```diff
@@ -131,4 +131,12 @@ struct GeoPluginConfigError {
             "Ensure the default search index is included in searchIndices."
         )
     }
+    
+    // MARK: - Tracker
+    static func trackerhDefaultNotFound(trackerName: String?) -> PluginError {
```
Comment (thisisabhash) [inline AmplifyPlugins/Geo/AWSLocationGeoPlugin/Configuration/GeoPluginConfigError.swift:136]: nit: `trackerDefaultNotFound`

**Review by thisisabhash — APPROVED** [review]
LGTM

**Pull request merged**